pyproject.toml

```diff
@@ -212,6 +213,156 @@ module = [
 ]
 disallow_any_generics = false
 
+[[tool.mypy.overrides]]
+module = [
+    "tests.__init__",
+    "tests.*.__init__",
+#    "tests.conftest",
+    # tests/
+    "tests.test_addnodes",
+    "tests.test_application",
+    "tests.test_errors",
+    "tests.test_events",
+    "tests.test_highlighting",
+    "tests.test_project",
+    "tests.test_quickstart",
+    "tests.test_roles",
+    "tests.test_search",
+    "tests.test_toctree",
+    "tests.test_versioning",
+#    "tests.utils",
+    # tests/test_builders
+#    "tests.test_builders.conftest",
```

Comment:
I think it's good to keep track of what is being changed by just commenting the whitelist. When nothing else is in this list, we can remove it in one go.

Reply:
Surely the non whitelisted modules are just implicitly "everything else"?

Reply:
Err... yes? but I'm not sure whether it's better to keep track of what was changed or if it's better to remove the line in each PR. I actually don't know which would actually trigger a merge conflict if two people work on two test modules separately (one would remove one line, the other another line)

Reply:
This seems relevant to https://github.com/sphinx-doc/sphinx/pull/12160#discussion_r1534138527 - I'm inclined to continue the comment-out-a-line-each-time approach, but I don't have a strong opinion on it.